components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImpl.java

```diff
@@ -2366,20 +2377,27 @@ public void updateApplicationByResourceId(String resourceId, ServiceProvider upd
                 throw buildClientException(APPLICATION_NOT_FOUND, msg);
             }
 
-            String updatedAppName = updatedApp.getApplicationName();
+            ServiceProvider storedSP = getServiceProvider(updatedSP.getApplicationID());
+            String updatedClaimConfigString = new Gson().toJson(updatedSP.getClaimConfig());
+            String storedClaimConfigString = new Gson().toJson(storedSP.getClaimConfig());
+            if (updatedClaimConfigString.equals(storedClaimConfigString)) {
+                isClaimConfigChanged = false;
+            }
```

Comment:
Can't we move this logic to the specific listener that requires to detect whether the claim configuration has changed?

Reply:
Addressed these comments in https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1963
And closing this PR.